File: C/练习/code_12_2.c
#include <stdio.h>


void is_prime(int *start, int *end) {
    int count = 0;
    for (*start; *start <= *end; (*start)++) {
        int i;
        for (i = 2; i < ((int)(*start / 2) + 1); i++) {
            if (*start % i == 0) {
                break;
            }
        }
        if (i >= ((int)(*start / 2) + 1)) {
            printf("%d ", *start);
            count++;
            if (count % 5 == 0) {
                printf("\n");
            }
        }
    }
    printf("\n");
}

int main(void) {
    int start = 101, end = 200;
    printf("素数在%d到%d之间的有：\n", start, end);
    is_prime(&start, &end);
    return 0;
}

File: C/Replicate_With_Python/code_12_2.py
def is_prime(start, end):
    count = 0
    for i in range(start, end + 1):
        for j in range(2, i // 2 + 1):
            if i % j == 0:
                break
        if j == i // 2:
            print(i, end=' ')
            count += 1
            if count % 5 == 0:
                print()
    print()
    return 0

if __name__ == '__main__':
    start, end = 101, 200
    is_prime(start, end)
